spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/environment/AwsSecretsManagerEnvironmentRepository.java

```diff
@@ -0,0 +1,162 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.config.server.environment;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+import com.amazonaws.services.secretsmanager.AWSSecretsManager;
+import com.amazonaws.services.secretsmanager.model.GetSecretValueRequest;
+import com.amazonaws.services.secretsmanager.model.GetSecretValueResult;
+import com.amazonaws.services.secretsmanager.model.ResourceNotFoundException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.springframework.cloud.config.environment.Environment;
+import org.springframework.cloud.config.environment.PropertySource;
+import org.springframework.cloud.config.server.config.ConfigServerProperties;
+import org.springframework.util.StringUtils;
+
+import static org.springframework.cloud.config.server.environment.AwsSecretsManagerEnvironmentProperties.DEFAULT_PATH_SEPARATOR;
+
+/**
+ * @author Tejas Pandilwar
+ */
+public class AwsSecretsManagerEnvironmentRepository implements EnvironmentRepository {
+
+	private static final Log log = LogFactory
+			.getLog(AwsSecretsManagerEnvironmentRepository.class);
+
+	private final ObjectMapper objectMapper;
+
+	private final AWSSecretsManager awsSmClient;
+
+	private final ConfigServerProperties configServerProperties;
+
+	private final AwsSecretsManagerEnvironmentProperties environmentProperties;
+
+	public AwsSecretsManagerEnvironmentRepository(AWSSecretsManager awsSmClient,
+			ConfigServerProperties configServerProperties,
+			AwsSecretsManagerEnvironmentProperties environmentProperties) {
+		this.awsSmClient = awsSmClient;
+		this.configServerProperties = configServerProperties;
+		this.environmentProperties = environmentProperties;
+		this.objectMapper = new ObjectMapper();
+	}
+
+	@Override
+	public Environment findOne(String application, String profileList, String label) {
+		final String defaultApplication = configServerProperties
+				.getDefaultApplicationName();
+		final String defaultProfile = configServerProperties.getDefaultProfile();
+
+		if (StringUtils.isEmpty(application)) {
+			application = defaultApplication;
+		}
+
+		if (StringUtils.isEmpty(profileList)) {
+			profileList = defaultProfile;
+		}
+
+		String[] profiles = StringUtils.commaDelimitedListToStringArray(profileList);
```

Comment:
it should be `StringUtils.trimArrayElements(StringUtils.commaDelimitedListToStringArray(profileList));` in order to support `profile-1, profile-2` currently implementation supports `profile-1,profile-2` with no space

Reply:
b54e1e7

Reply:
@eddumelendez thanks for the review. i addressed all your comments (b54e1e7) however the build is failing due to some rsocket pom dependency. can you please take a look? thanks

Reply:
@eddumelendez @spencergibb  did you get a chance to look at my above comment? will appreciate your help on resolving the build issue. thanks

Reply:
@tejas-pandilwar sorry for the delay. what if you run `./mvnw clean package -DskipTests -P spring,milestone`?

Reply:
@eddumelendez i ran the command locally and i see the same error as in the ci cirlce build.
`[ERROR]     Non-resolvable import POM: Could not find artifact io.rsocket:rsocket-bom:pom:1.1.0-SNAPSHOT in spring-snapshots (https://repo.spring.io/libs-snapshot-local) @ org.springfr
amework.boot:spring-boot-dependencies:2.4.0-SNAPSHOT, C:\Users\tejas\.m2\repository\org\springframework\boot\spring-boot-dependencies\2.4.0-SNAPSHOT\spring-boot-dependencies-2.4.0-SNAP
SHOT.pom, line 2674, column 19 -> [Help 2]`

Reply:
@eddumelendez @spencergibb did you get a chance to look at my earlier comment?

Reply:
@eddumelendez @spencergibb looks like the master build is also failing due to the same error. https://app.circleci.com/pipelines/github/spring-cloud/spring-cloud-config/169/workflows/f37154e3-84cd-442f-984d-3536fa186e7f/jobs/3790